### DISCUSSION TITLE
Parameter Sensitivity Study

### DISCUSSION THREAD

**Discussion by dmilson14** [pybamm-team/PyBaMM#3870] [Q&A]
Hello, 

Currently setting up a parameter sensitivity study to look at the impact of changing certain parameters within parameter values on the degradation (lithium plating concentration etc.) of the Marquis 2019 battery under different charging protocols. I have looked at updating the parameter and running a new simulation but this does take quite a lot of time. Is there a way to loop through the change in parameter values, I'll use dead lithium decay constant as an example, to then input them into a single simulation to be able to plot the various values at the end?

Any help would be greatly appreciated!

**Comment by martinjrobins** [comment]
Hi @dmilson14, `input_parameters` are what you need to do a parameter sweep, have a look under the "Input Parameters" heading on this page: https://train.oxrse.uk/material/HPCu/libraries/pybamm/03_parameter_values

Note that you can also pass in a list of dictionaries into the `solve` function of your solver, if you want to run the parameter sweep in parallel, see https://docs.pybamm.org/en/stable/source/api/solvers/base_solver.html#pybamm.BaseSolver.solve

**Reply by kumaryash7** [reply]
Hey, I am also trying to conduct a similar study. Can you share some snippet of your working code. It would be of great help for me.
TIA

**Reply by dmilson14** [reply]
Hello @martinjrobins, I have no doubt this is the answer to the problem but upon trying this by passing a list of dictionaries into the solve function to run this in parallel. I get the error 'list object has no attribute 'shape''. The common fix for this seems to be to convert it to a numpy array using np.array, however this simply crashes the kernel. I have installed and reinstalled numpy which still has not seemed to work. Is there any fix for this? The code I am using is below:

DLDC = [{'Dead lithium decay constant [s-1]': 1e-3}, {'Dead lithium decay constant [s-1]': 1e-4}]
        
sol = sim.solve(inputs={"Dead lithium decay constant [s-1]": DLDC})

**Reply by martinjrobins** [reply]
You just want a list of dictionaries (rather than a dict-of-list-of-dicts), so like this:

```python
DLDC = [{'Dead lithium decay constant [s-1]': 1e-3}, {'Dead lithium decay constant [s-1]': 1e-4}]
sol = sim.solve(inputs=DLDC)
```
